src/error/rescue.rs

```diff
@@ -130,7 +130,7 @@ impl Pipe for Sanitizer<'_> {
                 },
             );
 
-            Json(&payload).pipe(response.status(status_code))
+            Json(&payload).finalize(response.status(status_code))
```

Comment:
Delegation and branching feel less weird now. It also encourages the author to think about branching the way they would as if they were writing an extension trait for `ResponseBuilder`.

> Move as little as possible, hoist branching and pre conditions to the top of the finalize implementation.